File: ws/bonifaz/u2/WS24Zoo/Zoo/src/main/java/ec/edu/espe/zoo/model/Primate.java
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author Christian Bonifaz, Code Creators, DCCO-ESPE
 */
public abstract class Primate extends Mammal{

    public Primate(int id, String description, Date dateOnborn, char gender, int numberOfBones, Cage cage) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }

}

File: hw/villagomez/u2/hw18ZooJS/ec.edu.espe.zoo.model/Primate.js
const Mammal = require('./Mammal');

class Primate extends Mammal {

    constructor(id, description, dateOnborn, gender, numberOfBones, cage) {
        if (new.target === Primate) {
            throw new Error('Cannot instantiate abstract class Primate');
        }
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }
}

module.exports = Primate;